lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -72,6 +72,24 @@ CrossOp::inferReturnTypes(MLIRContext *context, std::optional<Location> loc,
   return success();
 }
 
+OpFoldResult EmitOp::fold(FoldAdaptor adaptor) {
+  // Return if the mapping is not the identity mapping.
+  int64_t numFields = cast<TupleType>(getInput().getType()).size();
+  int64_t numIndices = getMapping().size();
+  if (numFields != numIndices)
+    return {};
+  for (int64_t i = 0; i < numIndices; ++i) {
+    auto attr = getMapping()[i];
+    int64_t index = cast<IntegerAttr>(attr).getInt();
```

Comment:
Makes me wonder if we should be using dense array attr ...

Reply:
Yes! Done in #841.